source/material_model/rheology/friction_options.cc

```diff
@@ -0,0 +1,495 @@
+          }
+
+        if (this->simulator_is_past_initialization() && this->get_timestep_number() > 0 && in.requests_property(MaterialProperties::reaction_terms) && in.current_cell.state() == IteratorState::valid)
+          {
+            for (unsigned int j=0; j < volume_fractions.size(); ++j)
+              {
+                // q war n_evaluation_points
+                const double current_edot_ii =
+                  MaterialUtilities::compute_current_edot_ii (in.composition[q], ref_strain_rate,
+                                                              min_strain_rate, in.strain_rate[q], elastic_shear_moduli[j], use_elasticity,
+                                                              use_reference_strainrate, dte);
+
+                const unsigned int theta_position_tmp = this->introspection().compositional_index_for_name("theta");
+                const double theta_old = in.composition[q][theta_position_tmp];
+                const double current_theta = compute_theta(theta_old, current_edot_ii, cellsize);
+                const double theta_increment = current_theta - theta_old;
+
+                out.reaction_terms[q][theta_position_tmp] = theta_increment;
+              }
+          }
+      }
+
+      template <int dim>
+      std::pair<double,double>
+      FrictionOptions<dim>::calculate_depth_dependent_a_and_b(const Point<dim> &position, const int j) const
+      {
+        if ( a_and_b_source == Function )
+          {
+            Utilities::NaturalCoordinate<dim> point =
+              this->get_geometry_model().cartesian_to_other_coordinates(position, coordinate_system);
+
+            const double rate_and_state_parameter_a = rate_and_state_parameter_a_function->value(Utilities::convert_array_to_point<dim>(point.get_coordinates()),j);
+            const double rate_and_state_parameter_b = rate_and_state_parameter_b_function->value(Utilities::convert_array_to_point<dim>(point.get_coordinates()),j);
+
+            std::cout << " a is " << rate_and_state_parameter_a << " - and b is " << rate_and_state_parameter_b << std::endl;
+            return std::pair<double,double>(rate_and_state_parameter_a,
+                                            rate_and_state_parameter_b);
+          }
+        else if (a_and_b_source == None)
+          {
+            return std::pair<double,double>(0.0,0.0); // was return 1.0 in depth dependent for viscosity
+          }
+        else
+          {
+            Assert( false, ExcMessage("Invalid method for a and b depth dependence") );
+            return std::pair<double,double>(0.0,0.0);
+          }
+      }
+
+      template <int dim>
+      FrictionDependenceMechanism
+      FrictionOptions<dim>::
+      get_friction_dependence_mechanism() const
+      {
+        return friction_dependence_mechanism;
+      }
+
+      template <int dim>
+      bool
+      FrictionOptions<dim>::
+      get_theta_in_use() const
+      {
+        bool theta_in_use =false;
```

Comment:
-> = false

Reply:
is it then also ->=true when I update it? Like this: 

bool use_theta ->=false;
if (get_friction_dependence_mechanism() == rate_and_state_dependent_friction)
    use_theta ->=true;

Reply:
I should have explained better what I meant. With `->` I mean please change into whatever comes next. In this case this means a space between `=` and `false`.

Reply:
o true, I understood that in the other comments ;)